src/lib.rs

```diff
@@ -151,6 +151,44 @@ struct HelperInner {
     consumer_done: bool,
 }
 
+/// Error type for `from_env` function.
+#[derive(Debug)]
+pub enum ErrFromEnv {
+    /// There isn't env var, that describes jobserver to inherit.
+    IsNotConfigured,
+    /// Cannot connect following this process's environment.
+    PlatformSpecific {
+        /// Error.
+        err: io::Error,
```

Comment:
@weihanglo 
Could you try this interface in cargo before it's in a "stable" release of jobserver-rs?

It's possible that in practice we'll need to discern between different kinds of erros (e.g. "descriptors are closed" and everything else) to avoid bothering users with warnings too often.

This PR, however, merges all the errors into a single `io::Error` with multiple vague  `ErrorKind`s.
I'm not sure users will be able to use it reliably without relying on "string typing" (inspecting string descriptions returned by errors) or implementation details ("we know that closed descriptors are `ErrorKind::InvalidData` by reading source code of jobserver-rs").
(I don't like this part and I wish I read this PR in detail before its merge.)

Reply:
Sure. I'll try taking a look this week.

Reply:
> implementation details ("we know that closed descriptors are ErrorKind::InvalidData by reading source code of jobserver-rs").

Maybe jobserver can have the error kind documented on docs.rs?

Reply:
Sorry for the late response. I think the change works fine with Cargo. Cargo can just [ignore the jobserver if there are bad pipes](https://github.com/rust-lang/cargo/blob/7bf43f028ba5eb1f4d70d271c2546c38512c9875/src/cargo/util/config/mod.rs#L258-L262). No harm to get this patch released on Cargo side I believe :)